src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/CpsUtilities.cs

```diff
@@ -34,11 +34,15 @@ internal static class CpsUtilities
         public static string? ExtractAnalyzerFilePath(string projectDirectoryPath, string analyzerNodeCanonicalName)
```

Comment:
@drewnoakes Is this code needed anymore? There's a comment around line 52 or so that says "it is a newer form of the analyzer tree item's file path (VS16.7) which requires no processing." If it's just always that form then we can just delete this method which would be a nice cleanup.

Reply:
Apologies for the delay. I took a few days off around the weekend and have been catching up.

Yes, since 16.7 we simplified the identifiers used on dependencies such as this to remove the target framework and dependency "type" (e.g. `analyzerdependency`) from these strings. I believe the final checks here can be removed, though it may break compatibility with earlier SDK versions. You may like to test with earlier SDKs to be sure.

Reply:
> Apologies for the delay. I took a few days off around the weekend and have been catching up.

My policy is never apologize for taking vacation time. Apology rejected. :smile:

I'll file https://github.com/dotnet/roslyn/issues/50673 to track whether we can safely remove it.